language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/extensions/document/SyntaxTreeByNameTest.java

```diff
@@ -0,0 +1,95 @@
+    public void testFindFunctionInSameFile() throws Exception {
+        TestUtil.openDocument(serviceEndpoint, sameFile);
+        Range range = new Range(new Position(4, 13), new Position(4, 26));
+        BallerinaSyntaxTreeResponse syntaxTreeByNameResponse = LSExtensionTestUtil.getBallerinaSyntaxTreeByName(
+                sameFile.toString(), range, this.serviceEndpoint);
+        Assert.assertTrue(syntaxTreeByNameResponse.isParseSuccess());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getSyntaxTree());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getSource());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getDefFilePath());
+        TestUtil.closeDocument(this.serviceEndpoint, sameFile);
+    }
+
+    @Test(description = "Find ST by range in a project")
+    public void testFindFunctionInProject() throws Exception {
+        TestUtil.openDocument(serviceEndpoint, projectFile);
+        Range range = new Range(new Position(3, 12), new Position(3, 21));
+        BallerinaSyntaxTreeResponse syntaxTreeByNameResponse = LSExtensionTestUtil.getBallerinaSyntaxTreeByName(
+                projectFile.toString(), range, this.serviceEndpoint);
+        Assert.assertTrue(syntaxTreeByNameResponse.isParseSuccess());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getSyntaxTree());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getSource());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getDefFilePath());
+        TestUtil.closeDocument(this.serviceEndpoint, projectFile);
+    }
+
+    @Test(description = "Find ST by range in a module")
+    public void testFindFunctionInModule() throws Exception {
+        TestUtil.openDocument(serviceEndpoint, projectFile);
+        Range range = new Range(new Position(4, 11), new Position(4, 19));
+        BallerinaSyntaxTreeResponse syntaxTreeByNameResponse = LSExtensionTestUtil.getBallerinaSyntaxTreeByName(
+                projectFile.toString(), range, this.serviceEndpoint);
+        Assert.assertTrue(syntaxTreeByNameResponse.isParseSuccess());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getSyntaxTree());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getSource());
+        Assert.assertNotNull(syntaxTreeByNameResponse.getDefFilePath());
+        TestUtil.closeDocument(this.serviceEndpoint, projectFile);
+    }
+
+    @Test(description = "Not Find ST by range in a file")
+    public void testNotFindFunctionInSameFile() throws Exception {
+        TestUtil.openDocument(serviceEndpoint, sameFile);
+        Range range = new Range(new Position(6, 13), new Position(6, 26));
+        BallerinaSyntaxTreeResponse syntaxTreeByNameResponse = LSExtensionTestUtil.getBallerinaSyntaxTreeByName(
+                sameFile.toString(), range, this.serviceEndpoint);
+        Assert.assertFalse(syntaxTreeByNameResponse.isParseSuccess());
+        TestUtil.closeDocument(this.serviceEndpoint, sameFile);
+    }
```

Comment:
This has the same method name + description compared to the first test case (L42). Is that intentional?